lib/Sema/TypeCheckType.h

```diff
@@ -276,7 +276,7 @@ using OpenUnboundGenericTypeFn = llvm::function_ref<Type(UnboundGenericType *)>;
 
 /// A function reference used to handle a PlaceholderTypeRepr.
 using HandlePlaceholderTypeReprFn =
-    llvm::function_ref<Type(PlaceholderTypeRepr *)>;
+    llvm::function_ref<Type(ASTContext &, PlaceholderTypeRepr *)>;
```

Comment:
Yep I think that makes sense! I will pull this out into a separate PR and tag you on it.

Reply:
Opened as #36948. The commit is still in this PR (for the time being) since it's necessary for the changes, but once the other PR is merged I'll make sure it gets dropped from this PR.